File: 383-ransom-note/ransom-note.java
class Solution {
    public boolean canConstruct(String ransomNote, String magazine) {
        // if(ransomNote.length()>magazine.length())
        // {
        //     return false;
        // }
        Map<Character,Integer> map=new HashMap<>();
        for(int i=0;i<magazine.length();i++)
        {
           char c=magazine.charAt(i);
           map.put(c,map.getOrDefault(c,0)+1);
        }
        for(int i=0;i<ransomNote.length();i++)
        {
            char c=ransomNote.charAt(i);
            if(!map.containsKey(c)){
                return false;
            }
            map.put(c,map.get(c)-1);
        }
        for(int count:map.values())
        {
            if(count<0)
            {
                return false;
            }
        }
        return true;

    }
}

File: 383-ransom-note/ransom-note.py
class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        dict = collections.defaultdict(int)
        for c in magazine:
            dict[c] += 1
        for c in ransomNote:
            if c not in dict or dict[c] == 0:
                return False
            else:
                dict[c] -= 1
        return True
